### PR TITLE
refactor: use modern MUI api

### PR DESCRIPTION
**Pull request by SergeyKolchenko**
### Description

The latest Picasso version (26.8.0) uses a lot of deprecated MUI method/property calls. Every call emits a warning.
It is bearable for end users due to those warnings are just printed into console. But unit tests in some projects might fail because of `console.error` calls aren't expected there and Jest fails tests.

This PR start utilizing modern MUI props instead of deprecated ones. Pure refactoring was done, so, there must be no visual changes.

### How to test

* `PicassoProvider` uses `createTheme` instead of `createMuiTheme` due to the function was renamed.
* `Grid` uses `justifyContent` instead of `justify`
* `Loader` uses `determinate` variant instead of `static`
* `OutlinedInput` uses `minRows` and `maxRows` instead of `rows` and `rowsMax` respectively

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| Insert screenshots or screen recordings | Insert screenshots or screen recordings |

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

**Breaking change**

- ~codemod is created and showcased in the changeset~
- ~test alpha package of Picasso in StaffPortal~

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

### CODE REVIEW

**Review by TomasSlama — APPROVED** [review]
Nice!

**Comment by SergeyKolchenko** [comment]
@toptal-bot run all

**Review by augustobmoura — APPROVED** [review]
Thanks for the contribution @SergeyKolchenko! :bow:

**Review by ozgurkececioglu — APPROVED** [review]
Looking good 👍 thanks 🙇‍♂️

**Review by separatio — APPROVED** [review]
Tested. Checked the code. Looks good. :+1:

File: .changeset/dirty-ties-cover.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@toptal/picasso': patch
+'@toptal/picasso-provider': patch
+---
+
+Use modern MUI api to avoid warnings emmiting about deprecated props usage
```
Comment (separatio) [inline .changeset/dirty-ties-cover.md:6]: Please follow the changeset guidelines here: https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md#summary-of-the-change-changelog-message

**Review by dmaklygin — APPROVED** [review]
Checked the code. Looks good

**Pull request merged**